apis/management.cattle.io/v3/cluster_template_types.go

```diff
@@ -0,0 +1,62 @@
+package v3
+
+import (
+	"github.com/rancher/norman/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ClusterTemplate struct {
+	types.Namespaced
+
+	metav1.TypeMeta `json:",inline"`
+	// Standard object’s metadata. More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ClusterTemplateSpec `json:"spec"`
+}
+
+type ClusterTemplateSpec struct {
+	DisplayName         string `json:"displayName" norman:"required"`
+	Description         string `json:"description"`
+	Disabled            bool   `json:"disabled,omitempty"`
+	DefaultRevisionName string `json:"defaultRevisionName,omitempty" norman:"type=reference[clusterTemplateRevision]"`
+
+	Members       []Member `json:"members,omitempty"`
+	RevisionLimit int      `json:"revisionLimit,omitempty" norman:"default=10"`
+}
+
+type ClusterTemplateRevision struct {
+	types.Namespaced
+
+	metav1.TypeMeta `json:",inline"`
+	// Standard object’s metadata. More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the desired behavior of the the cluster. More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#spec-and-status
+	Spec ClusterTemplateRevisionSpec `json:"spec"`
+}
+
+type ClusterTemplateRevisionSpec struct {
+	Disabled            bool   `json:"disabled,omitempty"`
+	IsDefault           bool   `json:"isDefault,omitempty"`
+	ClusterTemplateName string `json:"clusterTemplateName,omitempty" norman:"type=reference[clusterTemplate],required"`
+	// Kubernetes version to use
+	Version       string         `yaml:"kubernetes_version" json:"kubernetesVersion,omitempty"`
```

Comment:
Have we decided on a top level version, or will we use k8s version from clusterConfig. RancherKubernetesEngineConfig?

Reply:
We can just use the clusterConfig.RancherKubernetesEngineConfig.Version. I will remove the extra parameter.